PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = ['account.bank.statement.line']
-    _inherit = ['hr.employee']
-    _inherit = ['pos.config']
-    _inherit = ["pos.order"]
-    _inherit = ["pos.payment"]
-    _inherit = ['pos.session']
-    _inherit = ['product.product']
-    _inherit = ['res.config.settings']

Total Changes: 17

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_hr/models/account_bank_statement.py
+++ b/addons/pos_hr/models/account_bank_statement.py
@@ -3,6 +3,6 @@ from odoo import fields, models
 
 
 class AccountBankStatementLine(models.Model):
-    _inherit = ['account.bank.statement.line']
+    _inherit = 'account.bank.statement.line'
 
     employee_id = fields.Many2one('hr.employee', string="Employee", help="The employee who made the cash move.")

--- a/addons/pos_hr/models/hr_employee.py
+++ b/addons/pos_hr/models/hr_employee.py
@@ -9,7 +9,7 @@ from odoo.tools import format_list
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     @api.model
     def _load_pos_data_domain(self, data):

--- a/addons/pos_hr/models/pos_config.py
+++ b/addons/pos_hr/models/pos_config.py
@@ -4,7 +4,7 @@ from odoo import models, fields, api
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     basic_employee_ids = fields.Many2many(
         'hr.employee', 'pos_hr_basic_employee_hr_employee', string="Employees with basic access",

--- a/addons/pos_hr/models/pos_order.py
+++ b/addons/pos_hr/models/pos_order.py
@@ -4,7 +4,7 @@ from markupsafe import Markup
 
 
 class PosOrder(models.Model):
-    _inherit = ["pos.order"]
+    _inherit = "pos.order"
 
     employee_id = fields.Many2one('hr.employee', string="Cashier", help="The employee who uses the cash register.")
     cashier = fields.Char(string="Cashier name", compute="_compute_cashier", store=True)

--- a/addons/pos_hr/models/pos_payment.py
+++ b/addons/pos_hr/models/pos_payment.py
@@ -2,7 +2,7 @@ from odoo import models, fields, api
 
 
 class PosPayment(models.Model):
-    _inherit = ["pos.payment"]
+    _inherit = "pos.payment"
 
     employee_id = fields.Many2one('hr.employee', string='Cashier', related='pos_order_id.employee_id', store=True, index=True)
 

--- a/addons/pos_hr/models/pos_session.py
+++ b/addons/pos_hr/models/pos_session.py
@@ -5,7 +5,7 @@ from odoo.tools import plaintext2html
 
 
 class PosSession(models.Model):
-    _inherit = ['pos.session']
+    _inherit = 'pos.session'
     employee_id = fields.Many2one(
         "hr.employee",
         string="Cashier",

--- a/addons/pos_hr/models/product_product.py
+++ b/addons/pos_hr/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class ProductProduct(models.Model):
-    _inherit = ['product.product']
+    _inherit = 'product.product'
 
     @api.model
     def _load_pos_data_fields(self, config_id):

--- a/addons/pos_hr/models/res_config_settings.py
+++ b/addons/pos_hr/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     # pos.config fields
     pos_basic_employee_ids = fields.Many2many(related='pos_config_id.basic_employee_ids', readonly=False,

--- a/addons/pos_hr/models/single_employee_sales_report.py
+++ b/addons/pos_hr/models/single_employee_sales_report.py
@@ -4,6 +4,7 @@ from odoo.osv.expression import AND
 
 
 class ReportPos_HrSingle_Employee_Sales_Report(models.AbstractModel):
+    _name = 'report.pos_hr.single_employee_sales_report'
     _inherit = ['report.point_of_sale.report_saledetails']
     _description = 'Session sales details for a single employee'
 
